include/deal.II/base/mpi_consensus_algorithms.h

```diff
@@ -1475,6 +1475,85 @@ namespace Utilities
                                      my_destinations.end()) ==
                   my_destinations.end());
         }
+
+
+
+        /**
+         * Handle exceptions inside the ConsensusAlgorithm::run() functions.
+         */
+        inline void
+        handle_exception(std::exception_ptr exception, const MPI_Comm &comm)
+        {
+#  ifdef DEAL_II_WITH_MPI
+          // an exception within a ConsensusAlgorithm likely causes an
+          // MPI deadlock. Abort with a reasonable error message instead.
+          try
+            {
+              std::rethrow_exception(exception);
```

Comment:
One of the tests also uncovered this issue:
```
/jenkins/workspace/dealii-tidy_PR-14364/include/deal.II/base/mpi_consensus_algorithms.h:1492:38: error: parameter 'exception' is passed by value and only copied once; consider moving it to avoid unnecessary copies [performance-
```

You're using `std::current_exception()` for the `exception` argument in the calling places. I think this might best be written in the following way, with an rvalue reference as argument:
```suggestion
        handle_exception(std::exception_ptr &&exception, const MPI_Comm &comm)
        {
#  ifdef DEAL_II_WITH_MPI
          // an exception within a ConsensusAlgorithm likely causes an
          // MPI deadlock. Abort with a reasonable error message instead.
          try
            {
              std::rethrow_exception(std::move(exception));
```